models/electric_relaxation/src/offline_evaluation/evaluate_model.py

```diff
@@ -0,0 +1,88 @@
+import numpy as np
+import pandas as pd
+from pathlib import Path
+import pickle
+import sys
+
+from sklearn.metrics import mean_squared_error, average_precision_score, roc_auc_score, brier_score_loss
+
+model_path = Path(__file__).resolve().parents[2] 
+sys.path.append(str(model_path))
+from configs.config_model import get_model_config
+
+
+def evaluate_model(model_config):
+    """
+    Evaluate a model's performance from the calibration partition.
+
+    This function loads a trained model from a pickle file, predicts outcomes for the calibration dataset,
+    calculates mean squared error (MSE), average precision, and Brier score for the predictions,
+    and saves the results as a dictionary in the artifacts folder.
+
+    Args:
+        model_config (dict): Configuration parameters for the model.
+
+    Returns:
+        None
+
+    Notes:
+    - There are 36 predictions for every true value (given 36 steps), so I am taking the mean value of them. Open to discuss this further
+    - Code for area under ROC curve didn't work, but leaving in for future development
+    - Should this also include test partition?
+    - When running the script, I get the following error from sklearn.metrics: No positive class found in y_true, recall is set to one for all thresholds.
+    - Instead of rewriting the code for metrics, it can be written as a loop or function in utils
+    """
+    print("Evaluating...")
+
+    df_calib = pd.read_parquet(model_path/"data"/"generated"/"calibration_predictions.parquet") 
+
+    steps = model_config["steps"]
+    depvar = [model_config["depvar"]] #formerly stepcols, changed to depvar to also use in true_values
+
+    for step in steps: 
+        depvar.append("step_pred_" + str(step))
+
+    df_calib = df_calib.replace([np.inf, -np.inf], 0)[depvar] 
+
+    pred_cols = [f"step_pred_{str(i)}" for i in steps] 
+    
+    df_calib["mse"] = df_calib.apply(lambda row: mean_squared_error([row["ged_sb_dep"]] * len(steps), #i tried replacing "ged_sb_dep" with depvar but it wouldn't work
```

Comment:
Did you use model_comfig['depvar'] instead of depvar? The hardcoded 'ged_sb_dep' is a point that I missed.

Reply:
I used depvar since I previously defined it as [model_config["depvar"]]. Now I just put in [model_config["depvar"]] instead of the hardcoded ged_sb_dep and it worked, though.